src/components/button.tsx

```diff
@@ -0,0 +1,25 @@
+import Link from "next/link";
+import React, { Children } from "react";
+
+interface Props {
+  url?: string;
+  clickFn?: () => void;
+  children: React.ReactNode;
+  btnclass: string;
+}
+
+const Button = ({ url, clickFn, children, btnclass }: Props) => {
```

Comment:
We will probably need to add forward Ref here to keep the element ref in re renders - we can look at it later